lisa/tools/tlb_stress.py

```
@@ -0,0 +1,1036 @@
+
+    def _verify_build_tools(self) -> None:
+        """Verify that essential build tools are available"""
+        essential_tools = ["gcc", "make"]
+        missing_tools = []
+
+        for tool in essential_tools:
+            result = self.node.execute(f"command -v {tool}", no_error_log=True)
+            if result.exit_code != 0:
+                missing_tools.append(tool)
+
+        if missing_tools:
+            raise FileNotFoundError(
+                f"Essential build tools missing: {', '.join(missing_tools)}"
+            )
+
+        # Verify C headers are available (optional for user-space programs)
+        header_test = self.node.execute(
+            "echo '#include <stdio.h>' | gcc -x c - -E -o /dev/null",
+            sudo=True,
+            no_error_log=True,
+        )
+        if header_test.exit_code != 0:
+            self._log.warning(
+                "C standard library headers (stdio.h) not available - "
+                "some advanced features may not work"
+            )
+            # Don't fail hard - many user-space programs can compile without
+            # full headers
+
+    def install(self) -> bool:
+        """Deploy TLB stress program to target nodes with bulletproof installation"""
+        try:
+            # Install build dependencies first
+            self._install_deps()
+
+            self._deploy_tlb_program()
+
+            # Verify installation was successful
+            if self._check_exists():
+                self._log.debug(
+                    f"TLB stress tool successfully installed at {self._bin}"
+                )
+                return True
+            else:
+                self._log.error(
+                    f"TLB stress installation failed - binary not found at {self._bin}"
+                )
+                return False
+
+        except Exception as e:
+            self._log.error(f"TLB stress installation failed with exception: {e}")
```

Let exception raise by itself.

---

done